integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/document_embedder.py

```diff
@@ -85,6 +87,32 @@ def __init__(
 
         self.backend: Optional[EmbedderBackend] = None
         self._initialized = False
+        self.is_hosted = urlparse(self.api_url).netloc in [
+            "integrate.api.nvidia.com",
+            "ai.api.nvidia.com",
+        ]
+        if self.is_hosted and not self.model:
+            # manually set default model
+            self.model = "NV-Embed-QA"
+
+    def default_model(self):
+        """Set default model in local NIM mode."""
+        valid_models = [
+            model.id for model in self.backend.models() if not model.base_model or model.base_model == model.id
+        ]
+        name = next(iter(valid_models), None)
+        if name:
+            warnings.warn(
+                f"Default model is set as: {name}. \n"
+                "Set model using model parameter. \n"
+                "To get available models use available_models property.",
+                UserWarning,
+                stacklevel=2,
+            )
+            self.model = self.backend.model = name
+        else:
+            error_message = "No locally hosted model was found."
+            raise ValueError(error_message)
```

Comment:
Looks like this code can be shared with all components? I'd suggest the following:
- Combine the `NimBackend` classes in `nvidia.embedders` and `nvidia.generators` into a single class (we no longer need the `Embedder/GeneratorBackend` classes) and move it to `haystack_integrations.util.nvidia`.
- Move the `default_model` method the combined class.
- Use the combined class for both the embedder and generator components.